src/Nethermind/Nethermind.Core/Collections/IReadOnlyListExtensions.cs

```diff
@@ -0,0 +1,16 @@
+using System.Collections.Generic;
+
+namespace Nethermind.Core.Collections;
+
+public static class IReadOnlyListExtensions
+{
+    public static IReadOnlyList<T> CappedTo<T>(this IReadOnlyList<T> readOnlyList, int length)
+    {
+        if (length > readOnlyList.Count)
+        {
+            return readOnlyList;
+        }
```

Comment:
The original type like `List<>` can still grow, so we shouldn't ever use this path

Reply:
Why? Its a read only list?